semantic-ast/src/Main.hs

```diff
@@ -12,22 +12,27 @@ import Data.ByteString (pack, readFile, ByteString)
 import System.IO (FilePath)
 import Options.Applicative hiding (style)
 import Data.Semigroup ((<>))
-import Text.Pretty.Simple (pPrint)
+import Text.Pretty.Simple (pPrint, pPrintNoColor)
+import Data.Foldable (for_)
 
 data SemanticAST = SemanticAST
   { format :: Format
-  , source :: Either FilePath Prelude.String
+  , color  :: Bool
+  , source :: Either [FilePath] Prelude.String
   }
 
+-- Usage: semantic-ast --format ARG [--no-color] [--sourceString STRING] [FILEPATHS…]
 parseAST :: Parser SemanticAST
 parseAST = SemanticAST
   <$> option auto
     ( long "format"
     <> help "Specify desired output: show, json, sexpression" )
-  <*> (Left <$> strOption
-      ( long "sourceFile"
-      <> metavar "FILEPATH"
-      <> help "Specify filepath containing source code to parse" )
+  <*> switch
+    ( long "no-color"
```

Comment:
The `no-color` option wasn't so much a critical CLI feature I felt was necessary, was more to just get comfortable seeing the pattern I need to reproduce when going from feature request -> Haskell code pipeline. The meta-lesson there being: start with the ADT, in this case `SemanticAST`, case on it below in `generateAST` which holds the logic for how to print something depending on the specified option and provided input, and then find an appropriate capability in `optparse-applicative` to properly parse that, including it in `parseAST`. In this case that's the `switch`.